core/variable_model.js

```diff
@@ -73,3 +73,18 @@ Blockly.VariableModel = function(name, opt_type, opt_id) {
 Blockly.VariableModel.prototype.getId = function() {
   return this.id_;
 };
+
+/**
+ * A custom compare function for the VariableModel objects.
+ * @param {Blockly.VariableModel} var1 First variable to compare.
+ * @param {Blockly.VariableModel} var2 Second variable to compare.
+ */
+Blockly.VariableModel.prototype.compareByName = function(var1, var2) {
```

Comment:
You can put this on Blockly.VariableModel instead of Blockly.VariableModel.prototype.